### ISSUE TITLE
Unsupported types in HLSL: float16_t2 and float16_t3

### ISSUE THREAD

**Issue by TheJackiMonster** [KhronosGroup/glslang#3093]
When trying to compile the shader from [here](https://github.com/GPUOpen-Effects/Hybrid-Shadows/blob/master/src/Shaders/filter_soft_shadows_pass_d3d12.hlsl) I got the following compilation error:

```
ERROR: 0:21: 'scalar or vector type' : Expected 
ERROR: 0:21: 'declaration' : Expected 
(21): error at column 22, HLSL parsing failed.
```

So these types (`float16_t2` and `float16_t3`) don't get treated as actual types by the compiler while `float16_t` exists. As a workaround it is possible to make a typedef before the actual code using them. But a proper solution would be better.

To clarify I have tried whether it just depends on the 16bit type feature but independent of me enabling it, the vector types are not treated as existing types.

**Comment by greg-lunarg** [comment]
Yes, unfortunately it appears that these types are not supported at this time.
